docs/contributing.md

```diff
@@ -58,6 +58,15 @@ for this purpose.
 
 To run a dag file in a temporary local environment, use `local-airflow.sh`. The script will symlink just the DAG provided to speed up parsing times.
 
+This requires Airflow to be installed locally. Ensure you have the appropriate Airflow version installed as defined in [cloud_composer_template.tf](/deployment/cloud_composer_template.tf#L126). For version 2.6.3:
+
+```
+VERSION=2.6.3
+pip install apache-airflow==$VERSION --constraint 'https://raw.githubusercontent.com/apache/airflow/constraints-${VERSION}/constraints-3.10.txt'
```

Comment:
Did this actually work for you? When I tried it after installing the `requirements.txt`, I had multiple conflicts with this version specifically.

Reply:
I don't think this option will work before airflow 2.8: `AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES_REGEXP`

Reply:
For airflow installation, do you think it's better we add it in [dev-requirement](https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/38812677c999b438b3b6f0bb0bf15de8ee407253/.github/dev-requirements.txt)?

Reply:
Yeah, this worked for me - I created a clean conda env with python 3.10 and ran this first, then `pip install -r .github/requirements.txt`. I'll try once more to verify if the order matters.

What does `AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES_REGEXP` do?

Reply:
This is required for plumbing data classes (particularly the SSH keys for QR tests) between steps. 

It was added in 2.8: `AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES_REGEXP`

Annoyingly, there was another flag that had the same behavior in prior versions:

- Old (regex syntax): https://airflow.apache.org/docs/apache-airflow/2.6.3/configurations-ref.html#allowed-deserialization-classes
- New (glob syntax): https://airflow.apache.org/docs/apache-airflow/2.8.2/configurations-ref.html#allowed-deserialization-classes

It's a breaking change in a patch version bump, so it's hard to reconcile.

Reply:
Okay, that makes sense. I'd like to keep the local airflow version the same as what runs in the composer env, since I expect people will just test their changes locally without a dev env going forward and subtle version differences could bite us if a broken change rolls through.

It seems like we can roll back to regex in `AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES` for 2.6.3, and I can look into how to make the install compatible with `requirements.txt`. What do you think?

Reply:
One other question, which Airflow version are we running in GitHub Actions? Will we need to change it to 2.6.3?

Reply:
I think the CI is installing the current version. That's what I get with `pip install -r requirements.txt`

Reply:
I'll revert this to 2.8 locally for now.